xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -193,27 +205,21 @@ bool CGUIWindowPVRGuide::Update(const std::string &strDirectory, bool updateFilt
 
 bool CGUIWindowPVRGuide::GetDirectory(const std::string &strDirectory, CFileItemList &items)
 {
-  bool bRefresh = false;
+  std::shared_ptr<CPVRRefreshTimelineItemsThread> refreshTimelineItemsThread;
 
   {
     CSingleLock lock(m_critSection);
 
-    // group change detected reset grid coordinates and refresh grid items
     if (!m_bRefreshTimelineItems && *m_cachedChannelGroup != *GetChannelGroup())
     {
-      CGUIEPGGridContainer* epgGridContainer = GetGridControl();
-      if (!epgGridContainer)
-        return true;
-
-      epgGridContainer->ResetCoordinates();
       m_bRefreshTimelineItems = true;
-      bRefresh = true;
+      refreshTimelineItemsThread = m_refreshTimelineItemsThread;
     }
   }
 
-  // never call RefreshTimelineItems with locked mutex!
-  if (bRefresh)
-    RefreshTimelineItems();
+  // never call DoRefresh with locked mutex!
+  if (refreshTimelineItemsThread)
+    refreshTimelineItemsThread->DoRefresh();
```

Comment:
I have to ensure that the thread object gets not destroyed while I'm not holding the lock! The point was not to get rid of the bool. This was just a side effect.